step/matcher/graph_pipeline.go

```diff
@@ -0,0 +1,58 @@
+package matcher
+
+import (
+	"regexp"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-steplib/bitrise-step-pull-intermediate-files/model"
+)
+
+type graphPipelineMatcher struct {
+	finishedWorkflows model.FinishedWorkflows
+	targetNames       []string
+	logger            log.Logger
+}
+
+func newGraphPipelineMatcher(finishedWorkflows model.FinishedWorkflows, targetNames []string, logger log.Logger) graphPipelineMatcher {
+	return graphPipelineMatcher{
+		finishedWorkflows: finishedWorkflows,
+		targetNames:       targetNames,
+		logger:            logger,
+	}
+}
+
+func (g graphPipelineMatcher) Matches() ([]string, error) {
+	var executedWorkflows []model.Workflow
+	for _, workflow := range g.finishedWorkflows {
+		if workflow.ExternalId == "" {
+			g.logger.Printf("Skipping workflow %s becaues it was not executed.", workflow.Name)
+			continue
+		}
+		executedWorkflows = append(executedWorkflows, workflow)
+	}
+
+	identifiers := make(map[string]bool)
+
+	if len(g.targetNames) == 0 {
+		for _, workflow := range executedWorkflows {
+			identifiers[workflow.ExternalId] = true
+		}
+
+		return convertKeySetToArray(identifiers), nil
+	}
+
+	for _, workflow := range executedWorkflows {
+		for _, target := range g.targetNames {
+			matched, err := regexp.MatchString(target, workflow.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			if matched {
+				identifiers[workflow.ExternalId] = true
+			}
```

Comment:
Shouldn't we break here? There's no reason to match against other targets I suppose if it was already a match.

BTW if we do break, we might not need the "set", we can just put (or not put) workflow external ids one by one in a slice.

Reply:
The break is a great idea. I will add it.

I want to keep the "set" just to be sure we are using every build id only once.